datacube/drivers/postgres/_api.py

```diff
@@ -934,7 +967,7 @@ def _from_expression(source_table, expressions=None, fields=None):
         if expressions:
             join_tables.update(expression.field.required_alchemy_table for expression in expressions)
         if fields:
-            join_tables.update(field.required_alchemy_table for field in fields)
+            join_tables.update(field.required_alchemy_table for field in fields if field)
```

Comment:
Direct follow-on from previous comment. It would be nice if we didn't have to do this check again.

Reply:
Same comment as above.